### DISCUSSION TITLE
Angular e mercado pago

### DISCUSSION THREAD

**Discussion by guilhermecostalopes** [frontendbr/forum#1909] [Perguntas e Respostas]
Angular.

https://www.mercadopago.com.br/developers/pt/guides/online-payments/checkout-api/receiving-payment-by-card

Coloquei este link `<script src="https://secure.mlstatic.com/sdk/javascript/v1/mercadopago.js"></script>,` no index.html

![70b84301d949a42a641fc91d8ac7970265143d95](https://user-images.githubusercontent.com/24395029/96726568-77200800-1388-11eb-95da-41c689a2296f.png)

Dá erro no **MercadoPago**, conforme imagem

```
window.Mercadopago.setPublishableKey("YOUR_PUBLIC_KEY");
```
![ad6755af36203c3a9b13eb97f086476cb80fcceb](https://user-images.githubusercontent.com/24395029/96726602-7e471600-1388-11eb-9a3a-57e83bd5d53d.png)

Está faltando algo ?

Procurei e não vi um npm, por exemplo

**Comment by raisiqueira** [comment]
@guilhermecostalopes Vc pode colocar um `(window as any).Mercadopago.createToken('', 'setCardTokenAndPay')`.

**Comment by guilhermecostalopes** [comment]
Mas este window, eu declaro ele aonde ?

**Comment by raisiqueira** [comment]
@guilhermecostalopes o Window é um objeto global do browser, não precisa declarar em nenhum lugar, caso sua aplicação esteja no lado do client vai ser reconhecida normalmente. Você pode fazer o teste direto no console do browser.

```js
window.Mercadopago
``` 

e ver o retorno que vai dar no console. o/

**Comment by guilhermecostalopes** [comment]
Estou usando Angular 9

Conforme imagem, no primeiro post

window.Mercadopago ou window.MercadoPago, dá erro

**Comment by raisiqueira** [comment]
@guilhermecostalopes conseguiu com o exemplo deste comentário? https://github.com/frontendbr/forum/issues/1909#issuecomment-713708210. O console do browser retorna algum erro? Em tese se vc acessar o objeto `window.Mercadopago.createToken('', 'setCardTokenAndPay')` no console do browser onde sua aplicação estar rodando vc tem que ver o retorno que a API do Mercado Pago vai te retornar.

**Comment by guilhermecostalopes** [comment]
@raisiqueira  pelo que entendi, está funcionando no console. Mas no sistema não


![image](https://user-images.githubusercontent.com/24395029/96787557-54addf00-13c8-11eb-9144-0171942af491.png)
